packages/core/src/components/tabs/tabs.tsx

```diff
@@ -160,20 +230,24 @@ export class Tabs extends AbstractPureComponent<ITabsProps, ITabsState> {
     }
 
     public componentDidUpdate(prevProps: ITabsProps, prevState: ITabsState) {
-        if (this.state.selectedTabId !== prevState.selectedTabId) {
-            this.moveSelectionIndicator();
-        } else if (prevState.selectedTabId != null) {
-            // comparing React nodes is difficult to do with simple logic, so
-            // shallowly compare just their props as a workaround.
-            const didChildrenChange = !Utils.arraysEqual(
-                this.getTabChildrenProps(prevProps),
-                this.getTabChildrenProps(),
-                Utils.shallowCompareKeys,
-            );
-            if (didChildrenChange) {
+        // adding a one second timeout here allows the overflowList to perform it's calculation
+        // so our ref has an accurate picture of the overflow state
+        setTimeout(() => {
+            if (this.state.selectedTabId !== prevState.selectedTabId) {
                 this.moveSelectionIndicator();
+            } else if (prevState.selectedTabId != null) {
+                // comparing React nodes is difficult to do with simple logic, so
+                // shallowly compare just their props as a workaround.
+                const didChildrenChange = !Utils.arraysEqual(
+                    this.getTabChildrenProps(prevProps),
+                    this.getTabChildrenProps(),
+                    Utils.shallowCompareKeys,
+                );
+                if (didChildrenChange) {
+                    this.moveSelectionIndicator();
+                }
```

Comment:
pretty sure we need some logic here to hide the indicator if one of the overflown tabs is selected.